src/OSPSuite.Core/Domain/UnitSystem/DimensionFactory.cs

```diff
@@ -24,18 +29,37 @@ public void AddDimension(IDimension dimension)
          _dimensions.Add(dimension.Name, dimension);
       }
 
-      public IEnumerable<IDimension> Dimensions => _dimensions;
-
-      public IEnumerable<string> DimensionNames => _dimensions.Keys;
-
       public IDimension Dimension(string name)
```

Comment:
Same behavior as before. It throws an exception if the dimension by name is not found